bot/exts/info/subscribe.py

```diff
@@ -178,22 +226,58 @@ async def cog_load(self) -> None:
     )
     async def subscribe_command(self, ctx: commands.Context, *_) -> None:  # We don't actually care about the args
         """Display the member's current state for each role, and allow them to add/remove the roles."""
-        button_view = RoleButtonView(ctx.author)
-        author_roles = [role.id for role in ctx.author.roles]
-        for index, role in enumerate(self.assignable_roles):
-            row = index // ITEMS_PER_ROW
-            button_view.add_item(SingleRoleButton(role, role.role_id in author_roles, row))
-
+        view = RoleButtonView(ctx.author, self.assignable_roles)
         await ctx.send(
             "Click the buttons below to add or remove your roles!",
-            view=button_view,
-            delete_after=DELETE_MESSAGE_AFTER,
+            view=view,
+            delete_after=DELETE_MESSAGE_AFTER
         )
 
+    async def _fetch_or_create_self_assignable_roles_message(self) -> tuple[discord.Message, discord.ui.View | None]:
+        """
+        Fetches the message that holds the self assignable roles view.
+
+        If the initial message isn't found, a new one will be created.
+        This message will always be needed to attach the persistent view to it
+        """
+        roles_channel: discord.TextChannel = await get_or_fetch_channel(constants.Channels.roles)
+
+        async for message in roles_channel.history(limit=30):
+            if message.content == self.SELF_ASSIGNABLE_ROLES_MESSAGE:
+                log.debug(f"Found self assignable roles view message: {message.id}")
+                return message, None
```

Comment:
Is there a reason we are returning `None` for the `discord.ui.View`? Couldn't we just use [`discord.ui.View.from_message`](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=ui%20button#discord.ui.View.from_message)?

Reply:
We can't really recreate the view using `from_message`, as this persistent view needs to be instantiated with the self assignable roles, and `from_message` cannot pick that up.